### ISSUE TITLE
2.10.0 fails to build on various architectures due to test failures

### ISSUE THREAD

**Issue by sebastic** [osmcode/libosmium#175]
Unfortunately the recent 2.10.0 release failed to build on various architectures, see:

 https://buildd.debian.org/status/package.php?p=libosmium

index_test_id_set failed on [armhf](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=armhf&ver=2.10.0-1&stamp=1478934804), [i386](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=i386&ver=2.10.0-1&stamp=1478929675), [mips](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=mips&ver=2.10.0-1&stamp=1478931961), [mipsel](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=mipsel&ver=2.10.0-1&stamp=1478931925), [hppa](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=hppa&ver=2.10.0-1&stamp=1478935026), [hurd-i386](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=hurd-i386&ver=2.10.0-1&stamp=1478957700), [kfreebsd-i386](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=kfreebsd-i386&ver=2.10.0-1&stamp=1478929707), [powerpc](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=powerpc&ver=2.10.0-1&stamp=1478929625) (32bit architectures):
```
33: Test command: /«PKGBUILDDIR»/build/test/index_test_id_set
33: Environment variables: 
33:  OSMIUM_TEST_DATA_DIR=/«PKGBUILDDIR»/test
33: Test timeout computed to be: 9.99988e+06
33: 
33: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
33: index_test_id_set is a Catch v1.5.8 host application.
33: Run with -? for options
33: 
33: -------------------------------------------------------------------------------
33: Iterating over IdSetDense
33: -------------------------------------------------------------------------------
33: /«PKGBUILDDIR»/test/t/index/test_id_set.cpp:48
33: ...............................................................................
33: 
33: /«PKGBUILDDIR»/test/t/index/test_id_set.cpp:73: FAILED:
33:   REQUIRE( it != s.end() )
33: with expansion:
33:   {?} != {?}
33: 
33: -------------------------------------------------------------------------------
33: Iterating over IdSetSmall
33: -------------------------------------------------------------------------------
33: /«PKGBUILDDIR»/test/t/index/test_id_set.cpp:130
33: ...............................................................................
33: 
33: /«PKGBUILDDIR»/test/t/index/test_id_set.cpp:162: FAILED:
33:   REQUIRE( *it == 1LL << 33 )
33: with expansion:
33:   8589934592 (0x200000000)
33:   ==
33:   8589934592 (0x200000000)
33: 
33: ===============================================================================
33: test cases:    6 |    4 passed | 2 failed
33: assertions: 1681 | 1679 passed | 2 failed
33: 
33/62 Test #33: index_test_id_set ........................***Failed    0.16 sec
test 34
      Start 34: index_test_file_based_index
```

geom_test_ogr_wkb also failed on [mips](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=mips&ver=2.10.0-1&stamp=1478931961), [s390x](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=s390x&ver=2.10.0-1&stamp=1478930675), [hppa](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=hppa&ver=2.10.0-1&stamp=1478935026), [powerpc](https://buildd.debian.org/status/fetch.php?pkg=libosmium&arch=powerpc&ver=2.10.0-1&stamp=1478929625) (big endian architectures):
```
28: Test command: /«PKGBUILDDIR»/build/test/geom_test_ogr_wkb
28: Environment variables: 
28:  OSMIUM_TEST_DATA_DIR=/«PKGBUILDDIR»/test
28: Test timeout computed to be: 9.99988e+06
28: 
28: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
28: geom_test_ogr_wkb is a Catch v1.5.8 host application.
28: Run with -? for options
28: 
28: -------------------------------------------------------------------------------
28: compare WKB point against GDAL/OGR
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:25
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:32: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: -------------------------------------------------------------------------------
28: compare WKB linestring against GDAL/OGR
28:   linestring
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:35
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:45: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: -------------------------------------------------------------------------------
28: compare WKB linestring against GDAL/OGR
28:   linestring, unique nodes, backwards
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:35
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:51: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: -------------------------------------------------------------------------------
28: compare WKB linestring against GDAL/OGR
28:   linestring, all nodes, forwards
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:35
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:57: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: -------------------------------------------------------------------------------
28: compare WKB linestring against GDAL/OGR
28:   linestring, all nodes, backwards
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:35
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:63: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: -------------------------------------------------------------------------------
28: compare WKB area against GDAL/OGR
28:   area_1outer_0inner
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:68
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:78: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:78: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:73: FAILED:
28: due to unexpected exception with message:
28:   basic_string::substr: __pos (which is 4294967295) > this->size() (which is
28:   103)
28: 
28: -------------------------------------------------------------------------------
28: compare WKB area against GDAL/OGR
28:   area_1outer_1inner
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:68
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:86: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:  @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:  @
28: 
28: 
28: 
28: 
28: 
28: 
28:   @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:  @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @ 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @ 
28: 
28: 
28: 
28: 
28: 
28: @
28:   
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @ 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: "
28: 
28: -------------------------------------------------------------------------------
28: compare WKB area against GDAL/OGR
28:   area_2outer_2inner
28: -------------------------------------------------------------------------------
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:68
28: ...............................................................................
28: 
28: /«PKGBUILDDIR»/test/t/geom/test_ogr_wkb.cpp:94: FAILED:
28:   REQUIRE( to_wkb(geometry.get()) == wkb )
28: with expansion:
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: -
28:   
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: -
28:   
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: $@
28: 
28: 
28: 
28: 
28: 
28: $@
28: 
28: 
28: 
28: 
28: 
28: &@
28: 
28: 
28: 
28: 
28: 
28: $@
28: 
28: 
28: 
28: -
28:   
28: 
28: &@
28: 
28: 
28: 
28: 
28: 
28: &@
28: 
28: 
28: 
28: 
28: 
28: $@
28: 
28: 
28: 
28: 
28: 
28: &@
28: 
28: 
28: 
28: 
28: 
28: $@
28: 
28: 
28: 
28: 
28: 
28: $@"
28:   ==
28:   "
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: -
28:   
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: -
28:   
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: @
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: @&
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: @&
28: 
28: -
28:   
28: 
28: 
28: 
28: @&
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: @&
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: @$
28: 
28: 
28: 
28: 
28: 
28: "
28: 
28: ===============================================================================
28: test cases:  3 |  3 failed
28: assertions: 10 | 10 failed
28: 
28/62 Test #28: geom_test_ogr_wkb ........................***Failed    1.19 sec
test 29
      Start 29: geom_test_projection
```

**Comment by sebastic** [comment]
I'm going to revert the Debian package back to 2.9.0, because the IdSet changes in 2.10.0 are also causing osmium-tool build failures. Both packages are affected by the ongoing transition to Boost 1.62, so I'd like to get those builds fixed to not block the transition.

**Comment by joto** [comment]
Should be fixed in 2.10.1. Please re-open if there is still a problem.
